GrocifyApp.UI/src/components/DefaultList.jsx

```diff
@@ -0,0 +1,108 @@
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+//Internal components
+import BaseModal from '../components/BaseModal';
+
+//Assets & Css
+import styles from './DefaultList.module.scss';
+
+//Consts
+import { GenericConsts } from '../consts/ENConsts';
+import ApiEndpoints from '../consts/ApiEndpoints';
+
+const DefaultList = ({ isOpen, onClose, child }) => {
+    const token = localStorage.getItem('token');
+    const [shoppingListData, setShoppingListData] = useState(null);
+    const [defaultShoppingList, setDefaultShoppingList] = useState(null);
+    const [oldDefaultShoppingList, setOldDefaultShoppingList] = useState(null);
+    const [isButtonDisabled, setButtonDisabled] = useState(true);
+
+    const handleSetDefaultList = (shoppingList) => {
+        setDefaultShoppingList(shoppingList);
+        setButtonDisabled(shoppingList.id === oldDefaultShoppingList?.id);
+    }
+
+    useEffect(() => {
+        const fetchData = async () => {
```

Comment:
can you check why useEffect is running twice?
maybe we need to do something like we've done in other place, having a variable firstRender and if has already pass this, it will not run this code again?